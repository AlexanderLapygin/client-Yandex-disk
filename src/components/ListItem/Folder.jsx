import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import icFolder from './ic_folder.svg';
import { deleteFolderRequest } from '../../actions/action';
import PropTypes from 'prop-types';

class Folder extends PureComponent {
  handleDelFolder = (e, path) => {
    this.props.deleteFolderRequest(path, this.props.location.pathname);
    e.stopPropagation();
  };

  render() {
    const { item } = this.props;
    return (
      <div className='d-flex justify-content-between'>
        <div>
          <img src={icFolder} alt='folder' className='folder_icon pr-3'/>
          {item.name}
        </div>
        <button onClick={(e) => this.handleDelFolder(e, item.path)}
                className='btn btn-outline-danger'>
          <i className='fas fa-times'/>
        </button>
      </div>

    );
  }
}

Folder.propTypes = {
  item: PropTypes.object,
};

const mapDispatchToProps = {
  deleteFolderRequest,
};

export default connect(null, mapDispatchToProps)(Folder);
