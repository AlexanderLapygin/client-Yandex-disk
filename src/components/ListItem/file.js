import React from 'react';
import icFile from './file.svg';

export const file = (item) => {
  return (
    <div className='d-flex justify-content-between'>
      <div className='item__file_name'>
        <img src={icFile} alt='folder'
             className='file_icon pr-3'/>
        {item.name}
      </div>
      <a href={item.file}>
        <button className='btn btn-outline-success'>
          <i className='fas fa-arrow-down'/>
        </button>
      </a>
    </div>
  );
};
