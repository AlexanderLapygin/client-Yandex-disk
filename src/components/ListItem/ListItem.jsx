import React, { PureComponent } from 'react';
import { connect } from 'react-redux';
import {withRouter} from 'react-router-dom';
import { getPath } from '../../selectors/selectors';
import {file} from './file'
import Folder from './Folder';


export class ListItem extends PureComponent {
  //при нажатии на папку так же изменяем путь адреса
  // и тем самым проваливаемся в неё
  handleClickFolder(type, name) {
    if (type === 'dir') {
      let newpath = this.props.currentPath.slice();
      newpath.shift();
      newpath.push(name);
      let newStrPath = '/' + newpath.join('/');
      this.props.history.push(newStrPath);
    }
  }

render(){
  const {item} = this.props;
  return(
    <li className='list-group-item  '
        onClick={() => this.handleClickFolder(item.type, item.name)}
    >
      {item.type === 'dir' ? <Folder item={item} location={this.props.history.location}/> : file(item)}
    </li>
  )
}
}

//добавляем данные из стора в пропсы
const mapStateToProps = state => {
  return ({
    currentPath: getPath(state),
  });
};

export default withRouter(connect(mapStateToProps, null)(ListItem))
