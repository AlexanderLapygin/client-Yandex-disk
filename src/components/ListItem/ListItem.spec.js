import React from 'react';
import { ListItem } from './ListItem';
import { shallow } from 'enzyme';

describe('List item test', () => {
  const props = {
    item: {
      name: 'filename',
      file: '',
    },
  };

  const ListItemComponent = shallow(<ListItem  {...props} />);

  it('snapshot', () => {
    expect(ListItemComponent).toMatchSnapshot();
  });
});
