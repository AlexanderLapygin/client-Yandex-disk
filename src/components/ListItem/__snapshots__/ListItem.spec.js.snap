// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List item test snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ListItem
    item={
      Object {
        "file": "",
        "name": "filename",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="d-flex justify-content-between"
      >
        <div
          className="item__file_name"
        >
          <img
            alt="folder"
            className="file_icon pr-3"
            src="file.svg"
          />
          filename
        </div>
        <a
          href=""
        >
          <button
            className="btn btn-outline-success"
          >
            <i
              className="fas fa-arrow-down"
            />
          </button>
        </a>
      </div>,
      "className": "list-group-item  ",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="item__file_name"
          >
            <img
              alt="folder"
              className="file_icon pr-3"
              src="file.svg"
            />
            filename
          </div>,
          <a
            href=""
          >
            <button
              className="btn btn-outline-success"
            >
              <i
                className="fas fa-arrow-down"
              />
            </button>
          </a>,
        ],
        "className": "d-flex justify-content-between",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="folder"
                className="file_icon pr-3"
                src="file.svg"
              />,
              "filename",
            ],
            "className": "item__file_name",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "folder",
                "className": "file_icon pr-3",
                "src": "file.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "filename",
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button
              className="btn btn-outline-success"
            >
              <i
                className="fas fa-arrow-down"
              />
            </button>,
            "href": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <i
                className="fas fa-arrow-down"
              />,
              "className": "btn btn-outline-success",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "fas fa-arrow-down",
              },
              "ref": null,
              "rendered": null,
              "type": "i",
            },
            "type": "button",
          },
          "type": "a",
        },
      ],
      "type": "div",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="d-flex justify-content-between"
        >
          <div
            className="item__file_name"
          >
            <img
              alt="folder"
              className="file_icon pr-3"
              src="file.svg"
            />
            filename
          </div>
          <a
            href=""
          >
            <button
              className="btn btn-outline-success"
            >
              <i
                className="fas fa-arrow-down"
              />
            </button>
          </a>
        </div>,
        "className": "list-group-item  ",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="item__file_name"
            >
              <img
                alt="folder"
                className="file_icon pr-3"
                src="file.svg"
              />
              filename
            </div>,
            <a
              href=""
            >
              <button
                className="btn btn-outline-success"
              >
                <i
                  className="fas fa-arrow-down"
                />
              </button>
            </a>,
          ],
          "className": "d-flex justify-content-between",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <img
                  alt="folder"
                  className="file_icon pr-3"
                  src="file.svg"
                />,
                "filename",
              ],
              "className": "item__file_name",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "alt": "folder",
                  "className": "file_icon pr-3",
                  "src": "file.svg",
                },
                "ref": null,
                "rendered": null,
                "type": "img",
              },
              "filename",
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <button
                className="btn btn-outline-success"
              >
                <i
                  className="fas fa-arrow-down"
                />
              </button>,
              "href": "",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <i
                  className="fas fa-arrow-down"
                />,
                "className": "btn btn-outline-success",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "fas fa-arrow-down",
                },
                "ref": null,
                "rendered": null,
                "type": "i",
              },
              "type": "button",
            },
            "type": "a",
          },
        ],
        "type": "div",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
